chalk-ir/src/fold.rs

```diff
@@ -55,20 +64,22 @@ pub use self::subst::Subst;
 /// ```rust,ignore
 /// let x = x.fold_with(&mut folder, 0);
 /// ```
-pub trait Folder<TF: TypeFamily>:
-    FreeVarFolder<TF> + InferenceFolder<TF> + PlaceholderFolder<TF> + TypeFolder<TF>
+pub trait Folder<TF: TypeFamily, TTF: TypeFamily>:
```

Comment:
Would it make sense to add a default here?
```Fold<TF: TypeFamily, TTF: TypeFamily = TF>```

Things like `Fold<ChalkIr>` read a bit better than `Fold<ChalkIr, ChalkIr>`, though I can also see that it could be confusing.

Reply:
I debated about it. It probably makes sense.